EPCData.API/IEpcDataApiCallerService.cs

```diff
@@ -0,0 +1,9 @@
+﻿using System.Collections.Generic;
+
+namespace EPCData.API
+{
+    public interface IEpcDataApiCallerService<out T>
+    {
+        IEnumerable<T> ExecuteRequestAsync(RequestParameters requestParameters);
```

Comment:
I just spotted a mistake here, method should return a Task if it is async. I have fixed this in the latest commit.